src/types.rs

```diff
@@ -226,22 +229,40 @@ impl PublicId {
         public_key : id.get_public_key(),
         public_sign_key : id.get_public_sign_key(),
         validation_token : id.get_validation_token(),
-        name : id.get_name()
+        name : id.get_name().clone(),
+        relocated : id.relocated,
       }
     }
 
+    pub fn name(&self) -> NameType {
+      self.name.clone()
+    }
+
     pub fn serialised_contents(&self)->Vec<u8> {
         let mut e = cbor::Encoder::from_memory();
         e.encode(&[&self]).unwrap();
         e.into_bytes()
     }
+
+    pub fn is_relocated(&self) -> bool {
```

Comment:
I asked for a calculated function here, not the implicit bool.  The preceding alternative was to have both name fields, but that has been discarded as an idea